post-docs/add-back-references.py

```diff
@@ -22,14 +22,30 @@
 from urllib.request import urlopen
 import semver
 
+log = logging.getLogger(__name__)
+
 airflow_redirects_link = "https://raw.githubusercontent.com/apache/airflow/main/docs/apache-airflow/redirects.txt"
 helm_redirects_link = "https://raw.githubusercontent.com/apache/airflow/main/docs/helm-chart/redirects.txt"
-providers_redirect_link = "populate-this"
 
 docs_archive_path = "../docs-archive"
 airflow_docs_path = docs_archive_path + "/apache-airflow"
 helm_docs_path = docs_archive_path + "/helm-chart"
-providers_docs_path = docs_archive_path + "/apache-airflow-providers"
+
+# if any new provider is added with redirects, add it to this list
```

Comment:
Since we are downloading the redirects from GitHub already can we simply loop over all directories in "docs-archive" and react to 404 not found by skipping a provider? that should be pretty fast.

Reply:
Why: the less chance we have to forget to update something, the better. This is for example why in my in-progress PR about sbom generation, I initially had a hard-coded list of Airflow versions - but eventually I settled on retriving them from tags in the git remote "apache/airflow" repo:
 https://github.com/apache/airflow/pull/31835/files#diff-54ea68b5e1e61ac0437df0ab7d99dc508595323b11847a957931e0d9007962d7R1241

Reply:
That was my initial idea too. But how do I get the list of directories in GitHub through my script?

Reply:
No need. Just list those in `docs-archive`

Reply:
Good idea. Missed that.
Yeah I agree with you on this, pushing a fix for that :)